### ISSUE TITLE
FrankenPHP Preview cache exception

### ISSUE THREAD

**Issue by sauls** [sulu/sulu#7332]
| Q | A
| --- | ---
| Sulu Version | 2.5.13
| PHP Version | 8.3.3
| DB Version | MariaDB 11.0.2
| Browser Version | Google Chrome Version 123.0.6312.58 (Official Build) (64-bit)

# Actual Behavior
I followed https://sulu.io/blog/running-sulu-with-frankenphp article, and created dual Kernel class. Locally when running on `dev` environment everyting works. But running `prod` environment the error is generated.
```
{"level":"error","ts":1711489705.64349,"msg":"PHP Fatal error:  Cannot declare class websiteWebspaceCollectionCache, because the name is already in use in /app/var/cache/preview/prod/sulu/websiteWebspaceCollectionCache.php on line 21","syslog_level":"err"}
```

# Expected Behavior

Error should not be generated.

# Steps to Reproduce

Running on [FrankenPHP](https://hub.docker.com/r/dunglas/frankenphp/tags) docker image with `prod` environemnt, if you clear cache and go to view published article. Then if you try to view same article in the admin preview you get the error and wise versa if after clearing cache first you open admin with preview, then navigating to article page will throw this error.

# Possible Solutions

Maybe this is from [FrankenPHP](https://frankenphp.dev/) also, how the cache is loaded into memory and generated, or not generated or differently generated per workers. :man_shrugging: 

Running same configuration with nginx + php there is no issues at all.

**Comment by alexander-schranz** [comment]
@sauls the issue seems to be here: https://github.com/sulu/sulu/blob/d635d8dcaa8679c4ff5ab3f1f52a6fa7ea3bc53a/src/Sulu/Bundle/CoreBundle/Resources/config/webspace.xml#L6

To avoid a conflict between Admin and Website Kernel we are using the `%sulu.context%`. But there is a third additional internal Kernel which is the PreviewKernel.

We require a compilerpass which check if it is the preview kernel `if (true === $container->getParameter('sulu.preview')) {` and prefix the `sulu_core.webspace.cache_class` with `preview`:

```
if (true === $container->getParameter('sulu.preview')) {
    $container->setParameter('sulu_core.webspace.cache_class', 'preview' + $container->getParameter('sulu_core.webspace.cache_class'));
}
```

@sauls do you want to give this fix a try?

**Comment by alexander-schranz** [comment]
Closing this as fixed in #7356
